src/java/org/apache/cassandra/service/accord/api/AccordRoutingKey.java

```diff
@@ -318,6 +319,119 @@ public long serializedSize(TokenKey key, int version)
         }
     }
 
+    // Allows the creation of a Range that is begin inclusive or end exclusive for a given Token
+    public static final class MinTokenKey extends TokenKey
+    {
+        private static final long EMPTY_SIZE;
+
+        @Override
+        public Range asRange()
+        {
+            AccordRoutingKey before = token.isMinimum()
+                                      ? new SentinelKey(table, true)
+                                      : new MinTokenKey(table, token.decreaseSlightly());
+
+            return new TokenRange(before, this);
+        }
+
+        static
+        {
+            EMPTY_SIZE = ObjectSizes.measure(new MinTokenKey(null, null));
+        }
+
+        public MinTokenKey(TableId tableId, Token token)
+        {
+            super(tableId, token);
+        }
+
+        public MinTokenKey withToken(Token token)
+        {
+            return new MinTokenKey(table, token);
+        }
+
+        @Override
+        public Token token()
+        {
+            return token;
+        }
+
+        @Override
+        public RoutingKeyKind kindOfRoutingKey()
+        {
+            return RoutingKeyKind.MIN_TOKEN;
+        }
+
+        @Override
+        public String suffix()
+        {
+            return token.toString();
+        }
+
+        public long estimatedSizeOnHeap()
+        {
+            return EMPTY_SIZE + token().getHeapSize();
+        }
+
+        public AccordRoutingKey withTable(TableId table)
+        {
+            return new MinTokenKey(table, token);
+        }
+
+        public static final MinTokenKey.Serializer serializer = new MinTokenKey.Serializer();
+        public static class Serializer implements AccordKeySerializer<MinTokenKey>
```

Comment:
this is copy/paste of `TokenKey.Serializer`, why not just extend that to avoid this duplication?  

```
public static class ExtendedTokenKey extends TokenKey
    {
        public static class Serializer extends TokenKey.Serializer implements AccordKeySerializer<ExtendedTokenKey>
        {
            @Override
            public void serialize(ExtendedTokenKey key, DataOutputPlus out, int version) throws IOException
            {
                super.serialize(key, out, version);
            }

            @Override
            public ExtendedTokenKey deserialize(DataInputPlus in, int version) throws IOException
            {
                return super.deserialize(in, version);
            }

            @Override
            public ExtendedTokenKey fromBytes(ByteBuffer bytes, IPartitioner partitioner)
            {
                return super.fromBytes(bytes, partitioner);
            }

            @Override
            public long serializedSize(ExtendedTokenKey key, int version)
            {
                return super.serializedSize(key, version);
            }
        }

        public ExtendedTokenKey(TableId tableId, Token token)
        {
            super(tableId, token);
        }
    }
```

The other option would be to make the `new` a factory so both can define

Reply:
I didn't extend because it seemed like it was different enough from a `TokenKey` that having it pass `instance` checks would be misleading and it also makes overriding methods that need to be different tricky.

I'll update them to share a serializer.

Reply:
Refactored to use a common `okenKeySerializer` class with a factory.